microservices/download/utils.js

```diff
@@ -1,4 +1,42 @@
-const superagent = require("superagent");
+const https = require("node:https");
+
+function doRequest(queryString) {
```

Comment:
I could move this new function I wrote "`doRequest()`" inside of the existing `findLink()` that uses it, so it's more obvious they're related and only ever used together. Makes the file flow a little easier to skim/read, too.

But it'd also mean it would need to be moved out again to be reused elsewhere in the file or exported or anything like that, I'm pretty sure.

So I'll leave this as-is unless a reviewer requests it. Ready for review now, IMO.

Reply:
I'm much more a fan of the Single Purpose Principle, I know I don't always follow it, but yeah I'm much more of a fan of keeping these separated.